tensorflow/dtensor/mlir/utils/collective_lowering.cc

```diff
@@ -1383,9 +1383,6 @@ struct DTensorAllReduceLowering
 struct DTensorReduceScatterLowering
     : public impl::DTensorReduceScatterLoweringBase<
           DTensorReduceScatterLowering> {
-  void getDependentDialects(mlir::DialectRegistry& registry) const override {
```

Comment:
Actually, it turns out that we do need the dependent dialects because these ops uses LayoutAttr, which is only defined in the DTensorDiaglect (ran into a test failure without this in reduce_scatter lowering). I'll add these back (add also add them to other collective lowering passes, as they most likely will eventually need them too) in the internal CL before submitting it.

Reply:
Thanks for investigating that!